### ISSUE TITLE
Bazel requires java8 cask which doesn't work anymore

### ISSUE THREAD

**Issue by uri-canva** [Homebrew/homebrew-core#36049]
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] are reporting a bug others will be able to reproduce and not asking a question or requesting software. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh. To get software added or changed in Homebrew please file a [Pull Request](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:
- What you were trying to do (and why)
`brew install bazel` to install bazel
- What happened (include command output)
```
$ brew install bazel
bazel: Java 1.8 is required to install this formula.
JavaRequirement unsatisfied!
You can install with Homebrew Cask:
 brew cask install homebrew/cask-versions/java8
You can download from:
 https://www.oracle.com/technetwork/java/javase/downloads/index.html
Error: An unsatisfied requirement failed this build.
$ brew cask install zulu8
==> Satisfying dependencies
==> Downloading https://cdn.azul.com/zulu/bin/zulu8.33.0.1-jdk8.0.192-macosx_x64.dmg
######################################################################## 100.0%
==> Verifying SHA-256 checksum for Cask 'zulu8'.
==> Installing Cask zulu8
==> Running installer for zulu8; your password may be necessary.
==> Package installers may write to any location; options such as --appdir are ignored.
installer: Package name is Zulu 8.33.0.1
installer: Installing at base path /
installer: The install was successful.
🍺  zulu8 was successfully installed!
$ brew install bazel
bazel: Java 1.8 is required to install this formula.
JavaRequirement unsatisfied!
You can install with Homebrew Cask:
 brew cask install homebrew/cask-versions/java8
You can download from:
 https://www.oracle.com/technetwork/java/javase/downloads/index.html
Error: An unsatisfied requirement failed this build.
$ brew cask install java8
Updating Homebrew...
==> Caveats
This Cask makes minor modifications to the JRE to prevent issues with
packaged applications, as discussed here:

  https://bugs.eclipse.org/bugs/show_bug.cgi?id=411361

If your Java application still asks for JRE installation, you might need
to reboot or logout/login.

Installing java8 means you have AGREED to the license at
  https://www.oracle.com/technetwork/java/javase/terms/license/index.html

==> Satisfying dependencies
==> Downloading https://download.oracle.com/otn-pub/java/jdk/8u192-b12/750e1c8617c5452694857ad95c3ee230/jdk-8u192-macosx-x64.dmg
==> Downloading from https://download.oracle.com/otn-pub/java/jdk/8u192-b12/750e1c8617c5452694857ad95c3ee230/jdk-8u192-macosx-x64.dmg?AuthParam=1547598517_1e2cc235c24649f6aab99227ef75e13b

curl: (22) The requested URL returned error: 404 Not Found
Error: Download failed on Cask 'java8' with message: Download failed: https://download.oracle.com/otn-pub/java/jdk/8u192-b12/750e1c8617c5452694857ad95c3ee230/jdk-8u192-macosx-x64.dmg
```
- What you expected to happen
Bazel is installed
- Step-by-step reproduction instructions (by running `brew install` commands)
`brew install bazel`

**Comment by uri-canva** [comment]
```
$ brew gist-logs bazel
https://gist.github.com/b2f177a7eca62c35a6a3529c0a1d42d9```

Note the logs relate to the tap formula not the core formula, even though the core formula was the one I tried to install. Maybe because missing dependencies cause it to fail before starting to build?

**Comment by uri-canva** [comment]
Note that even though the `java8` cask not working is a separate issue, already captured here https://github.com/Homebrew/homebrew-cask-versions/pull/6847, since Oracle Java 8 isn't available anymore it's likely that it won't be able to be fixed. Instead the Bazel formula should accept `zulu8` as a substitute, or not require java at all, like the official formula: https://github.com/bazelbuild/homebrew-tap/blob/42a4bf1f4411084321e95f89c0653fe62f74ae10/Formula/bazel.rb

**Comment by SMillerDev** [comment]
https://github.com/Homebrew/homebrew-core/pull/35357#issuecomment-449598950 bazel themselves aren't very clear on what version they need. If they officially support 8 or higher we could switch to that instead but keeping track of alternatives like zulu won't likely happen.

**Comment by uri-canva** [comment]
They officially support 8 and 9. It's complicated because there's three jdks involved: the jdk you use to build Bazel, the jdk you use to run Bazel and the jdk you use to build with Bazel.

|JDK   |8|9|10|11|
|------|-|-|--|--|
|Build |✅|✅|❌|❌|
|Run   |✅|✅|✅|✅|
|Target|✅|✅|⚠️|⚠️|

**Comment by fxcoudert** [comment]
Java 8 is working again in Cask: https://github.com/Homebrew/homebrew-cask-versions/pull/6853
